core/network/src/on_demand.rs

```diff
@@ -457,8 +453,8 @@ impl<B, E> OnDemandCore<B, E> where
 			let mut request = self.pending_requests.pop_front().expect("checked in loop condition; qed");
 			request.timestamp = Instant::now();
 			trace!(target: "sync", "Dispatching remote request {} to peer {}", request.id, peer);
-
-			service.execute_in_context(|ctx| ctx.send_message(peer, request.message()));
+			let sender = self.network_sender.as_ref().expect("OnDemand should have a network sender at dispatch");
```

Comment:
I think it's unnecessary because the early-exit would only be hit in the tests, and it's now easy enough to add the sender as part of the test setup, see https://github.com/paritytech/substrate/pull/1340/files#diff-aef9e236f4fcf5cb038d15657c1d0382R645

Actually, the channel is now used in test assertions so you need to set the sender for the tests to pass...